alf/environments/suite_carla.py

```diff
@@ -835,6 +874,19 @@ def __init__(self,
             success_reward (float): the reward for arriving the goal location.
             success_distance_thresh (float): success is achieved if the current
                 location is with such distance of the goal
+            collision_reward (float): the reward for collision. Note that this
+                reward is only given once at the first step of contiguous collisions.
+            max_stuck_at_collision_seconds (float): the episode will end and is
+                considerred as failure if the car is stuck at the collision for
+                so many seconds,
+            stuck_at_collision_distance (float): the car is considerred as being
+                stuck at the collision if it is within such distance of the first
+                collision location.
```

Comment:
What if the car hits some other car with a speed so that the ending stuck position is this distance away from the first collision location?

Reply:
Right now this is not handled because there are other reasons of getting stuck at some place which cannot be easily distinguished from this. Currently another observed reason for getting stuck is that the waypoints are behind the car. In this situation, the car does not know to turn around or to go backward. But I feel I should just let it try to learn even it is stuck.